src/cart/crt_rpc.c

```diff
@@ -1745,10 +1751,14 @@ crt_rpc_common_hdlr(struct crt_rpc_priv *rpc_priv)
 
 	if (crt_rpc_cb_customized(crt_ctx, &rpc_priv->crp_pub) &&
 	    !crt_opc_is_swim(rpc_priv->crp_req_hdr.cch_opc)) {
+		/* Corresponding decref in crt_handle_rpc() */
+		RPC_ADDREF(rpc_priv);
```

Comment:
[Not requesting changes] I find it unclear why we need to take an extra reference here: we are not duplicating the rpc_priv pointer, and cc_rpc_cb is not supposed to release the crp_pub reference passed to it (cc_rpc_cb doesn't even know that the rpc_hldr_arg argument is a crp_pub reference). Is it because some cc_rpc_cb implementations have bugs?

Reply:
My thinking, and perhaps incorrect, was that not knowing what the pointer is cc_rpc_cb might duplicate it/store it for a deferred processing without a knowledge of needing to addref it first, so we are doing it for it.

Reply:
but could cc_rpc_cb processing also have some part being executed asynchronously, like cc_iv_resp_cb ??

Reply:
@bfaccini am not sure as I don't know full daos implementation of those, @wangdi1 what do you think?